lumen/app/Jobs/ImportEvent.php

```diff
@@ -27,28 +27,26 @@ public function __construct($event)
     public function handle(MongoClient $client)
     {
         $eventStorage = $client->test->events;
-
         $toImport = $this->event;
-
         $event = $eventStorage->findOne([
             'external_id' => data_get($toImport, 'external_id')
-        ]); 
+        ]);
 
-        if (empty($event)) {
+        if(empty($event)) {
             $venueStorage = $client->test->venues;
             $venueId = data_get($toImport, 'metadata.venue_id');
-            
+
             $venue = $venueStorage->findOne([
-                'external_id' => $venueId
+                'external_id' => data_get($toImport, 'metadata.venue_id')
             ]);
 
-            if (!empty($venue)) {
+            if(!empty($venue)) {
                 data_set($toImport, 'venue', $venue);
                 return $eventStorage->insertOne($toImport);
             }
-
             dispatch(new ImportVenue($venueId));
-            dispatch(new ImportEvent($toImport));
+
+            dispatch(new ImportEvent($this->toImport));
```

Comment:
We're getting an error here because the variables is never defined.

```
In ImportEvent.php line 49:

  Undefined property: App\Jobs\ImportEvent::$toImport

```